lib/lm3s/usart.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2013 Gareth McMullin <gareth@blacksphere.co.nz>
+ * Copyright (C) 2015 Daniele Lacamera <root at danielinux.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/lm3s/usart.h>
+
+
+void usart_set_baudrate(uint32_t usart, uint32_t baud)
+{
+    /* TODO */
+    (void)usart;
+    (void)baud;
+}
+
+void usart_set_databits(uint32_t usart, int bits)
+{
+    /* TODO */
+    (void)usart;
+    (void)bits;
+}
+
+void usart_set_stopbits(uint32_t usart, enum usart_stopbits sb)
+{
+    /* TODO */
+    (void)usart;
+    (void)sb;
+}
+
+void usart_set_parity(uint32_t usart, enum usart_parity par)
+{
+    /* TODO */
+    (void)usart;
+    (void)par;
+}
+
+void usart_set_mode(uint32_t usart, enum usart_mode mode)
+{
+    /* TODO */
+    (void)usart;
+    (void)mode;
+}
+
+void usart_set_flow_control(uint32_t usart, enum usart_flowcontrol fc)
+{
+    /* TODO */
+    (void)usart;
+    (void)fc;
+}
+
+void usart_enable(uint32_t usart)
+{
+	(void)usart;
+}
+
+void usart_disable(uint32_t usart)
+{
+	(void)usart;
+}
+
```

Comment:
just don't even implement this stuff.  Functions that appear to all exist and if you don't happen to go all the way into the source you don't notice that they are actually no-ops aren't helping anyone.